v8js_class.cc

```diff
@@ -925,6 +927,14 @@ static void v8js_script_dtor(zend_resource *rsrc) /* {{{ */
 }
 /* }}} */
 
+// Convert an interned or non-interned string into a persistent string. Works around interned strings being cleaned up automatically.
+static zend_string* persistent_string_from_maybe_interned_string(zend_string *string) { /* {{{ */
+	if (ZSTR_IS_INTERNED(string)) {
```

Comment:
What about removing the `ZSTR_IS_INTERNED` test and calling `zend_string_init` unconditionally? After all the `zend_string_dup`, which you're calling otherwise, simply does the reverse check and then call `zend_string_init`:

```c
static zend_always_inline zend_string *zend_string_dup(zend_string *s, int persistent)
{
    if (ZSTR_IS_INTERNED(s)) {
        return s;
    } else {
        return zend_string_init(ZSTR_VAL(s), ZSTR_LEN(s), persistent);
    }   
}
```